app/controllers/challenges_controller.rb

```diff
@@ -124,6 +129,10 @@ def create
   end
 
   def edit
+    if !logged_in? || @challenge.creator_id.nil? || challenge.creator_id != current_user.id
```

Comment:
Technically speaking, you don't need @challenge.creator_id.nil?
Also, it should be @challenge.creator_id, not challenge.creator_id.
Lastly, we might want to handle the "you're not the creator of the challenge" case differently. Maybe a new page similar to /redirect saying "you do not have permission to access this." or something.

Reply:
> Technically speaking, you don't need @challenge.creator_id.nil

I included that because it was in the original `challenges/edit` template and I didn't re-assess it. I don't suppose it was necessary there, either? But yeah, I'll remove it.

> Also, it should be @challenge.creator_id, not challenge.creator_id.

Oh man, that's bad. I tested the *submissions* edit page as a different user, and since the logic was the same, I assumed this would also work, because I'm used to languages where that sort of typo would be a compile-time error. Is there a linter or something I can run to catch bugs like that?

> Maybe a new page similar to /redirect saying "you do not have permission to access this." or something.

I edited the error page template (in the original commit) so that it will tell you that you need to log in, or if you're the wrong user, that you're unauthorized.